vendor/frida/CMakeLists.txt

```diff
@@ -5,11 +5,15 @@ if(NOT WIN32)
 endif()
 
 if (APPLE)
+  set(ARCHITECTURE_PATH x86_64 )
+  if(CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL "arm64")
```

Comment:
Use `CMAKE_SYSTEM_PROCESSOR`. When cross-compiling, the target architecture can be different than the host architecture.